presto-hive/src/main/java/com/facebook/presto/hive/metastore/BridgingHiveMetastore.java

```diff
@@ -177,6 +177,26 @@ public void renameColumn(String databaseName, String tableName, String oldColumn
         alterTable(databaseName, tableName, table);
     }
 
+    @Override
+    public void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        Optional<org.apache.hadoop.hive.metastore.api.Table> source = delegate.getTable(databaseName, tableName);
+        if (!source.isPresent()) {
+            throw new TableNotFoundException(new SchemaTableName(databaseName, tableName));
+        }
+        org.apache.hadoop.hive.metastore.api.Table table = source.get();
+        if (table.getSd().getCols().size() == 1) {
+            throw new PrestoException(NOT_SUPPORTED, "Dropping last column is not supported");
```

Comment:
I simply moved the validation to `HiveMetadata`. But I'm not sure it is enough for your change request.